src/rp2040/hardware_structs/include/hardware/structs/interp.h

```diff
@@ -8,17 +10,34 @@
 #define _HARDWARE_STRUCTS_INTERP_H
 
 #include "hardware/address_mapped.h"
-#include "hardware/platform_defs.h"
 #include "hardware/regs/sio.h"
 
+// reference to datasheet: https://datasheets.raspberrypi.org/rp2040/rp2040-datasheet.pdf#tab-registerlist_sio
+// BITMASK : FIELDNAME [BITRANGE] (RESETVALUE): DESCRIPTION
+
 typedef struct {
+    _REG_(SIO_None_OFFSET)
     io_rw_32 accum[2];
+
+    _REG_(SIO_None_OFFSET)
     io_rw_32 base[3];
+
+    _REG_(SIO_INTERP0_POP_LANE0_OFFSET)
     io_ro_32 pop[3];
+
+    _REG_(SIO_INTERP0_PEEK_LANE0_OFFSET)
     io_ro_32 peek[3];
+
+    _REG_(SIO_None_OFFSET)
     io_rw_32 ctrl[2];
+
+    _REG_(SIO_INTERP0_ACCUM0_ADD_OFFSET)
     io_rw_32 add_raw[2];
-    io_wo_32 base01;
+
+    _REG_(SIO_INTERP0_BASE_1AND0_OFFSET)
+    // On write, the lower 16 bits go to BASE0, upper bits to BASE1 simultaneously
+    io_rw_32 base01;
```

Comment:
Page 56 of the datasheet says that the `INTERP0_BASE_1AND0` register is "WO" - would it make sense to retain the `io_wo_32 base01;` here?

Reply:
Oh, interesting. The SVD file says read write, so one of them is wrong xD

Reply:
Oh! Hmmm....

Reply:
Yeah, that was a bug in the SVD which #544 fixes! Whoops.

Reply:
My soul can rest knowing I just have to run the generator again and everything will be fixed, rather than having to edit the stucts manually to reflect this xD.

Reply:
Note this is fixed in develop branch of SDK, but not in the new SVD in the latest doc release